### PR TITLE
[DO NOT MERGE] New nsconfig option for enabling HMR by default

### PR DESCRIPTION
**Pull request by NickIliev**
Related to [https://github.com/NativeScript/nativescript-cli/pull/4411](https://github.com/NativeScript/nativescript-cli/pull/4411)

To be merged after CLI v.53 is released.

### CODE REVIEW

File: docs/start/cli-basics.md [hunk 1]
```diff
@@ -238,6 +239,49 @@ The `run` command will again take a few seconds, as the NativeScript CLI will be
...
+tns run
+```
+
+- `useLegacyWorkflow` is set to `false`: The option disables the default livesync, in favor of the enhanced HMR (no need to pass additional flag).
+- `useLegacyWorkflow` is set to `true`(default value): The option enables the default livesync experience (explicit enabling of HMR via `--hmr` flag is still possible).
+
+
+> **Note:** There are conceptual differences when the `useLegacyWorkflow` is used in **debug** build versus when used in **release** builds. When the flag is enabled against a debug build, the extended HMR experience is enabled. However, in cases where the `--release` flag is passed (release build) then only the `--bundle` is executed in the background (No HMR). The reason for this is to avoid bringing additional HMR logic and files into production. For details about the differences between Webpack livesync and HMR refer to [the comparative table](/performance-optimizations/bundling-with-webpack#hot-module-replacement).
+
+In case, when you have enabled the default HMR via the `useLegacyWorkflow` option, but you need to test without using HMR you can pass the `--no-hmr` flag to disable the feature for the current build.
```
Comment (Fatme) [inline docs/start/cli-basics.md]: `no-bundle` provides the default experience
`no-hmr` provides webpack build without hmr

File: docs/start/cli-basics.md [hunk 2]
```diff
@@ -238,6 +239,49 @@ The `run` command will again take a few seconds, as the NativeScript CLI will be
 
 Now that you have a full setup in place for developing your apps let’s look at a few additional things you can do.
 
+### Using RUN with HMR
+
+The developer experience significantly improves by using the [HMR (Hot Module Replacement)](/performance-optimizations/bundling-with-webpack#hot-module-replacement) feature introduced in NativeScript 5. The option requires Webpack (enabled by default in all NativeScript 5.x.x templates) and behind the scene is creating Webpack bundle with the enhancement of Hot Module Replacement developer experience.
```
Comment (Fatme) [inline docs/start/cli-basics.md]: `The option requires nativescript-dev-webpack plugin` (enabled by default in all NativeScript 5.x.x templates) maybe?

**Review by tsonevn — APPROVED** [review]

**Comment by tsonevn** [comment]
test

**Review by Fatme — APPROVED** [review]

**Comment by NickIliev** [comment]
test

**Pull request merged**